src/core_modules/capture-core-utils/RulesEngine/VariableService/VariableService.js

```diff
@@ -8,24 +8,24 @@ import getDateUtils from '../dateUtils/dateUtils';
 import momentConverter from '../converters/momentConverter';
 
 import type {
-    ProgramRulesContainer,
     ProgramRuleVariable,
-    EventData,
+    InputEvent,
     EventsDataContainer,
     OptionSets,
     DataElement,
     DataElements,
     TrackedEntityAttribute,
     TrackedEntityAttributes,
-    Entity,
     Enrollment,
     Constants,
     OrgUnit,
     DateUtils,
+    RuleVariable,
 } from '../rulesEngine.types';
+import type { Entity } from '../../../capture-core/components/Relationships/relationships.types';
```

Comment:
Looked into this one. the type Entity should be replaced by TEIValues from rulesEngine.types.